examples/plot_openfmri_ds001.py

```diff
@@ -0,0 +1,214 @@
+                                         'sub{0:03d}'.format(sub_id),
+                                         'anatomy', 'highres001.nii.gz')
+    _subject_data['func_events'] = _run_events
+    _subject_data['onset_files'] = all_event_files
+
+    return Bunch(**_subject_data)
+
+
+def _openfrmri_contrasts(ref_design_matrix, dataset_dir, model_id, task_id):
+    # Build basic contrast_matrix.
+    contrast_matrix = np.eye(ref_design_matrix.shape[1])
+    contrasts = dict([(column, contrast_matrix[i])
+                      for i, column in enumerate(ref_design_matrix.columns)])
+
+    # Load information from openfmri files
+    task = 'task{0:03d}'.format(task_id)
+    condkey_file = os.path.join(dataset_dir, 'models',
+                                'model{0:03d}'.format(model_id),
+                                'condition_key.txt')
+    cond_df = pd.read_csv(condkey_file,
+                          sep=OTHER_FILE_SEP,
+                          header=None)
+    info_conditions = cond_df[cond_df[0] == task]
+    conds_name = info_conditions[2].tolist()
+    contrasts_file = os.path.join(dataset_dir, 'models',
+                                  'model{0:03d}'.format(model_id),
+                                  'task_contrasts.txt')
+    contrasts_df = pd.read_csv(contrasts_file,
+                               sep=OTHER_FILE_SEP,
+                               header=None)
+    task_contrasts = contrasts_df[contrasts_df[0] == task]
+
+    # Complete contrast specification
+    for idx, crow in task_contrasts.iterrows():
+        contrasts[crow[1]] = np.zeros(ref_design_matrix.shape[1])
+        for widx, weight in enumerate(crow[2:]):
+            contrasts[crow[1]] += weight*contrasts[conds_name[widx]]
+
+    return contrasts
+
+# dataset parameters
+sub_id = 1
+model_id = 1
+task_id = 1
+
+# download openfmri dataset
+dataset_dir = fetch_openfmri_dataset(dataset_id='ds001')
+
+# fetch openfmri dataset
+data = _glob_openfmri_data(dataset_dir, sub_id, model_id, task_id)
+
+# construct design matrices
+design_matrices = []
+contrasts = []
+for session in range(len(data.func_events)):
+    onsets = []
+    durations = []
+    names = []
+    n_cond = len(data.func_events[session]['conditions'])
+    for c in range(n_cond):
+        onsets += data.func_events[session]['onsets'][c]
+        durations += data.func_events[session]['durations'][c]
+        n_events = len(data.func_events[session]['durations'][c])
+        names += [data.func_events[session]['conditions'][c]] * n_events
+    paradigm = pd.DataFrame({'onset': onsets, 'duration': durations,
+                             'name': names})
+    n_scans = nib.load(data.func[session]).get_data().shape[3]
+    frame_times = np.linspace(0, (n_scans - 1) * data.TR, n_scans)
+    drift_model = 'Cosine'
+    hrf_model = 'Canonical With Derivative'
+    design_matrix = make_design_matrix(frame_times, paradigm,
+                                       hrf_model=hrf_model,
```

Comment:
you could specify `hrf_model` and `drift_model` only here. It's not canonical design, but examples are meant to be read easily.